storage-access.bs

```diff
@@ -169,46 +206,55 @@ ISSUE(10): Remove step 3.9 if we determine that nested <{iframe}>s should be abl
-1. Let |has storage access| (a [=boolean=]) be the result of running a UA-defined set of steps to determine if |doc| (a [=third party=] in |topDoc|) has access to its [=first party data=].
+1. Let |has storage access| (a [=boolean=]) be the result of running an [=implementation-defined=] set of steps to determine if |key|'s [=partitioned storage key/embedded site=] has access to its [=first party data=] on |key|'s [=partitioned storage key/top-level site=].
 1. If |has storage access| is true, set |flag set|'s [=has storage access flag=].
+1. [=Save the storage access flag set=] for |key| in |map|.
 1. Return |has storage access|.
 
-To <dfn type="abstract-op">determine the storage access policy</dfn> for {{Document|Documents}} |doc| and |topDoc| with {{Promise}} |p|, run these steps:
+To <dfn type="abstract-op">determine the storage access policy</dfn> for [=partitioned storage key=] |key| with {{Document}} |doc| and {{Promise}} |p|, run these steps:
+
+1. Let |implicitly granted| and |implicitly denied| (each a [=boolean=]) be the result of running an [=implementation-defined=] set of steps to determine if |key|'s [=partitioned storage key/embedded site=]'s request for storage access on |key|'s [=partitioned storage key/top-level site=] should be granted or denied without prompting the user.
+1. If |implicitly granted| is true, [=/resolve=] |p| and return.
+1. If |implicitly denied| is true, [=/reject=] |p| and return.
+1. Ask the user if they would like to grant |key|'s [=partitioned storage key/embedded site=] access to its [=first party data=] when it's loaded in a [=third party context=] on |key|'s [=partitioned storage key/top-level site=], and wait for an answer. Let |expressly granted| and |expressly denied| (both [=booleans=]) be the result.
 
-1. Assert: |topDoc| is the [=active document=] of |doc|'s [=Document/browsing context=]'s [=top-level browsing context=].
-1. Let |implicitly granted| and |implicitly denied| (each a [=boolean=]) be the result of running a UA-defined set of steps to determine if |doc|'s request for storage access on |topDoc| should be granted or denied without prompting the user.
-1. If |implicitly granted| is true, [=resolve=] |p| and return.
-1. If |implicitly denied| is true, [=reject=] |p| and return.
-1. Ask the user if they would like to grant |doc| access to its [=first party data=] when it is a [=third party=] on |topDoc|, and wait for an answer. Let |expressly granted| and |expressly denied| (both [=booleans=]) be the result.
-1. If |expressly granted| is true, [=resolve=] |p| and return. <!-- WebKit's DocumentStorageAccess.cpp#L191 -->
-1. Let |flag set| be the result of [=obtain the storage access flag set|obtaining the storage access flag set=] of |doc|.
+    Note: While |expressly granted| and |expressly denied| cannot both be true, they *can* both be false: users can dismiss the prompt without choosing to allow or deny the request. This is interpreted as a denial.
```

Comment:
This is true in general but I believe WebKit's implementation is a modal dialog in which the user has no "Cancel" option.

Reply:
There's nothing here requiring implementations to be able to produce the both false case.

Reply:
@johnwilander, I've adjusted the wording of this note in 6c1e4b4 to capture this better.